agnocast_heaphook/src/lib.rs

```diff
@@ -510,4 +510,37 @@ mod tests {
 
         unsafe { free(ptr) };
     }
+
+    #[test]
+    fn test_calloc_normal() {
+        // Arrange
+        let start = MEMPOOL_START.load(Ordering::SeqCst);
+        let end = MEMPOOL_END.load(Ordering::SeqCst);
+        let elements = 4;
+        let element_size = 256;
+        let calloc_size = elements * element_size;
+        
+        // Act
+        let ptr = calloc(elements, element_size);
+        
+        // Assert
```

Comment:
Could you add an assertion whether the start address is properly stored?
```
    // store `start_addr`
    let start_addr_ptr: *mut usize = (aligned_addr - POINTER_SIZE) as *mut usize;
    unsafe { *start_addr_ptr = start_addr };
```

Reply:
@Enigamict 
I meant the functional correctness of `start_addr`.
Please refer to https://github.com/tier4/agnocast/blob/main/docs/heaphook_alignment.png

Reply:
Sorry, I was wrong 🙇